### PR TITLE
Fix OSC52

### LINKED ISSUE #54: osc52 does not work
Connecting to remote linux box via ssh running tmux and latest tmux-thumbs.
Doing `echo -e "\033]52;c;$(base64 <<< hello)\a"` gets `hello` into the system's clipboard which is running the terminal, so OSC52 works with the terminal.
When enabling OSC52 in tmux-thumbs the copied data does not get propagated back to the system's clipboard, but the following gets printed (if copying 1.1.1.1): `Ptmux;]52;0;MS4xLjEuMQ==\`

Edit:
tmux `next-3.3`
tmux-thumbs: 3415117b

### PR DESCRIPTION
**Pull request by pbartyik**
to fix #54 
once solution for tmux/tmux#2363 gets merged this should be merged too.

I decided to make use of the now empty `fn send_osc52()`

change to `pane_command` is result of formatter

### CODE REVIEW

**Comment by fcsonline** [comment]
Thanks for all this investigation. My terminal doesn't support is 52 and I can't test it properly.

When the tmux patch gets merged, I think we won't need any specific code to handle this.

The user will be able to configure the `-t` flag on their `thumbs-command`:

https://github.com/fcsonline/tmux-thumbs#thumbs-command

What do you think?

**Comment by pbartyik** [comment]
yeah that is how I initially did the workaround too. The problem I encountered was when I did
```
tmux set-buffer -w {} && tmux display-message \"Copied {}\"
```
the buffer did not get set. Once I turned it around it started working
```
tmux display-message \"Copied {}\" && tmux set-buffer -w {}
```
This is why I thought it was better to keep it as a separate command to avoid possible issues.

Then again.. proper documentation can help :)
I don't think backwards compatibility would be an issue here as I doubt this ever worked.

**Comment by pbartyik** [comment]
@fcsonline tmux patch is merged. Would you like to add/remove something to/from this?

**Comment by fcsonline** [comment]
@isundaylee can you try it?

**Comment by fcsonline** [comment]
There is a is 52 reference in `tmux-thumbs.sh`

File: README.md [hunk 1]
```diff
@@ -183,6 +182,14 @@ For example:
 set -g @thumbs-command 'echo -n {} | pbcopy'
 ```
 
+OSC52:
+
+Depends on tmux git (at least fc9cdc)  and on `set -s set-clipboard on|external` in tmux config.
```
Comment (fcsonline) [inline README.md:187]: For old versions of tmux, can we suggest a command like?:

```
set -g @thumbs-command 'tmux set-buffer {} | echo -e "\033]52;c;$(base64 <<< {})\a"'
```

Comment (fcsonline) [reply README.md:187]: Can we point `fc9cdc` to the original commit, please?

Comment (pbartyik) [reply README.md:187]: > For old versions of tmux, can we suggest a command like?:
> 
> ```
> set -g @thumbs-command 'tmux set-buffer {} | echo -e "\033]52;c;$(base64 <<< {})\a"'
> ```
won't work :( when doing `tmux run blah` `blah` does not run in terminal see: https://github.com/tmux/tmux/issues/2363#issuecomment-685432373

> Can we point fc9cdc to the original commit, please?
>

I have no idea how to do that but let me figure it out

Comment (fcsonline) [reply README.md]: > won't work :( when doing tmux run blah blah does not run in terminal see: tmux/tmux#2363 (comment)
Ouch true :sweat_smile: 

> I have no idea how to do that but let me figure it out

I can add it later

**Comment by fcsonline** [comment]
It could be cool to know the opinion of @isundaylee

**Comment by isundaylee** [comment]
Hi - sorry I have been a bit busy this week and haven't got a chance to test/look into this yet. I'll try to get to it this weekend.

**Comment by pbartyik** [comment]
any luck?

**Comment by isundaylee** [comment]
Apology for the delay- I just got some time to look into this.

With a Git-built `tmux` and `@thumbs-command` set to `set-buffer -w`, it works as expected.

Do we have a workaround for older version of `tmux`? I saw the suggestion above of setting `@thumbs-command` to echo the OSC52 escape sequence, but it sounds like that wouldn't work since "when doing tmux run blah blah does not run in terminal". Do we have any other workaround?

**Comment by pbartyik** [comment]
Not one I'm aware of, otherwise I'd have gone that route instead of asking tmux to be patched updated :)

**Comment by isundaylee** [comment]
Hmm - would this mean that we won't be able to use OSC 52 at all without the latest `tmux` (without having to go into copy mode and manually select/copy the match)? Personally I run `tmux-thumbs` on many different servers, some with self-built `tmux` binary, and some with `tmux` from the system install that I don't have control over.

In any case,  @fcsonline feel free to go ahead and merge. I'm currently running a fork with some personal additions so I'll be fine.

Thanks!

**Comment by pbartyik** [comment]
> Hmm - would this mean that we will no longer be able to use OSC 52 at all without the latest `tmux` (without having to go into copy mode and manually select/copy the match)? Personally I run `tmux-thumbs` on many different servers, some with self-built `tmux` binary, and some with `tmux` from the system install that I don't have control over.

Does OS52 work on those instances, if so how?

**Comment by isundaylee** [comment]
Yes - for some reason printing escape chars currently still work for me (i.e. the `thumbs-osc52` option). I'm running this branch (https://github.com/isundaylee/tmux-thumbs/commits/mine) but I think all OSC52-related changes have been upstreamed already.

I wonder if it only works for some tmux versions. I have used 2.8 and next-3.1 with it. For reference I'm using iTerm 3.3.9. Just tested macOS' default Terminal - works there as well.

**Comment by pbartyik** [comment]
I wonder if tmux used to send stuff to terminal in `tmux run blah` and stopped after a version?
If so we should not merge this and leave osc52 as is. Users of old tmux could use `thumbs` to do this, users of new would go via the `set buffer -w` method.

**Comment by isundaylee** [comment]
Yeah I'm curious too and could dig into this a bit this weekend. `next-3.3` was the tmux version that didn't work right? I'll try that out this weekend and see if it's a tmux version issue or something else.

**Comment by isundaylee** [comment]
Just got a chance to test it with next-3.3 (commit `92a2e7411f26257bba32317793eac3d42a330b6b` to be specific) - the OSC52 escape copy seems to still work for me.

I'm running this custom branch of `tmux-thumbs` - but as I mentioned I believe all my OSC52-related changes have already been upstreamed.

**Comment by pbartyik** [comment]
closing this request as there is a need for osc escape sequence to stay.
If for whatever reason the escape sequence doesn't work for the user, the user can go via the custom thumbs command.

**Pull request closed without merging**